CINERGI Result Detail View

arottersman opened the pull request:
## Overview

Flesh out the cinergi specific detail view

[Template here](https://docs.google.com/document/d/1k9_I1gp9V8F2ieL-LfN2OYCimrrqyf9IKAGtCSlM8Uw/edit?usp=sharing)

Branched off #2425, commits for this work are f586fa3 and f230751

@jfrankl, here's another you may want to style review

Connects #2393 

### Demo

![screen shot 2017-10-25 at 1 54 32 pm](https://user-images.githubusercontent.com/7633670/32016821-ddac8678-b992-11e7-9fde-adcc4f12a322.png)

![screen shot 2017-10-25 at 1 54 40 pm](https://user-images.githubusercontent.com/7633670/32016839-ec66f2e8-b992-11e7-87a2-6bd3a2355cc7.png)


### Notes
#### Many organizations
There can be more organizations than fit in a single page. Is it useful to display them all, or should they be capped at a certain threshold?
<details>
<summary>pic</summary>

![jauqolzvnr](https://user-images.githubusercontent.com/7633670/32016241-62c6ce4c-b991-11e7-8431-7bf8cf691c18.gif)

</details>

#### Duplicate organizations
<details>
<summary>pic</summary>

<img width="460" alt="screen shot 2017-10-25 at 1 58 15 pm" src="https://user-images.githubusercontent.com/7633670/32016312-979296e2-b991-11e7-9db2-a0547a4b1cee.png">

</details>


## Testing Instructions

 * Pull and `bundle`
 * Use the City Of Philadelphia-Schuylkill river HUC-12 and search "water epa"
 * The `EPA Office of Water (OW) STORET...` result has a complete view with all fields
 * click pretty much any other result and confirm partially filled views render okay, too

rajadain commented:
Taking a look now.

rajadain commented:
We should probably add something like this:

```diff
diff --git a/src/mmw/js/src/data_catalog/views.js b/src/mmw/js/src/data_catalog/views.js
index 49daf9f..564d9c3 100644
--- a/src/mmw/js/src/data_catalog/views.js
+++ b/src/mmw/js/src/data_catalog/views.js
@@ -486,6 +486,7 @@ var ResultDetailsBaseView = Marionette.LayoutView.extend({
     },
 
     closeDetails: function() {
+        window.closePopover();
         this.model.collection.closeDetail();
     }
 });
```

to prevent this:

![2017-10-26 12 01 48](https://user-images.githubusercontent.com/1430060/32063613-9f188340-ba45-11e7-8056-c8665ab0c8ab.gif)

rajadain commented on src/mmw/js/src/data_catalog/templates/resultDetailsCinergi.html:
```diff
@@ -4,27 +4,111 @@
+                <th>Organizations</th>
+                <td>
+                    {% for org in contact_organizations %}
+                         {{ org }}<br />
```
Maybe instead of a `<br />` we should wrap these in a `<p>` so that it is easier to tell when a long name is wrapping vs a new name starting:

![image](https://user-images.githubusercontent.com/1430060/32064058-cf1cc17c-ba46-11e7-9ad6-e3fa33a98e91.png)

arottersman commented:
- Added a fixup for the `<p>` tag instead of using `<br />`
![screen shot 2017-10-26 at 12 21 55 pm](https://user-images.githubusercontent.com/7633670/32064839-038b6ce0-ba49-11e7-80cd-03e6c9dc3e21.png)

- Added a commit for closing the popup when closing the detail view.

Will wait for Emilio's feedback for any additional commits.

rajadain commented:
Great. Here's a diff of what I did to make the final screenshot in https://github.com/WikiWatershed/model-my-watershed/issues/2393#issuecomment-339739696:

```diff
diff --git a/src/mmw/js/src/data_catalog/templates/resultDetailsCinergi.html b/src/mmw/js/src/data_catalog/templates/resultDetailsCinergi.html
index 44a7f2f..3bc354e 100644
--- a/src/mmw/js/src/data_catalog/templates/resultDetailsCinergi.html
+++ b/src/mmw/js/src/data_catalog/templates/resultDetailsCinergi.html
@@ -11,19 +11,23 @@
                 <th>Contacts</th>
                 <td>
                     {% for contact in contact_people %}
-                         {{ contact }}<br />
+                         <p>{{ contact }}</p>
+                    {% else %}
+                        None provided
                     {% endfor %}
                 </td>
             </tr>
             <tr>
-            <tr>
                 <th>Organizations</th>
                 <td>
-                    {% for org in contact_organizations %}
-                         {{ org }}<br />
+                    {% for org in orgs %}
+                         <p>{{ org }}</p>
+                    {% else %}
+                        None provided
                     {% endfor %}
                 </td>
             </tr>
+            <tr>
                 <th>Data&nbsp;collected&nbsp;between</th>
                 <td>
                     {% if begin_date %}
diff --git a/src/mmw/js/src/data_catalog/views.js b/src/mmw/js/src/data_catalog/views.js
index 49daf9f..a2aa31c 100644
--- a/src/mmw/js/src/data_catalog/views.js
+++ b/src/mmw/js/src/data_catalog/views.js
@@ -494,8 +494,11 @@ var ResultDetailsCinergiView = ResultDetailsBaseView.extend({
     template: resultDetailsCinergiTmpl,
 
     templateHelpers: function() {
-        var topicCategories = this.model.get('resource_topic_categories');
+        var topicCategories = this.model.get('resource_topic_categories'),
+            contactOrgs = this.model.get('contact_organizations');
+
         return {
+            'orgs': contactOrgs ? contactOrgs.filter(utils.distinct) : null,
             'top_categories': this.model.topCinergiCategories(20),
             'resource_topic_categories_str': topicCategories ?
                 topicCategories.join(", ") : null,
```

Also heads up that this will need to be rebased.

arottersman commented:
Thanks for working that out with them. This has been rebased on develop. I've pushed another fixup with your diff. Should be ready for a final look.

rajadain commented:
Taking another look now.

rajadain commented on src/mmw/js/src/data_catalog/templates/resultDetailsCinergi.html:
```diff
@@ -11,19 +11,24 @@
                     {% endfor %}
                 </td>
             </tr>
             <tr>
```
Extra `<tr>`

jfrankl commented:
Just taking a look. Looks like these sections can get lengthy, which could get overwhelming, especially if several sections are long.

One solution would be to show the first and provide the user to expand the sections they find interesting. Like this:

![image](https://user-images.githubusercontent.com/1809908/32074543-bee0dc8c-ba67-11e7-961c-c3db47878324.png)

@arottersman @rajadain @ajrobbins let me know what you think of this direction?

ajrobbins commented:
I like that treatment, and think we should do it for the contact field as well. I'd punt it to another card, but probably most efficient to add it here due to turbo pteradactyl.

arottersman commented:
Added a commit for the truncate button.
![screen shot 2017-10-26 at 6 40 19 pm](https://user-images.githubusercontent.com/7633670/32080431-9041e090-ba7d-11e7-9611-0fba71b0849f.png)

arottersman commented:
Rebased on develop.

rajadain commented:
Taking a look now.

arottersman commented:
Thanks for reviewing, and resolving all those unknowns!
